src/prometheus/definition.go

```diff
@@ -701,3 +724,41 @@ func getRawEntityID(parentGroupLabel, groupLabel, entityID string, groups defini
 	}
 	return rawEntityID, nil
 }
+
+func WithLabelValueInMetricNameAndWithLabelValueMappedToMetricValue(rawMetricName, label, format, valueLabel string, valueMap map[string]int) definition.FetchFunc {
```

Comment:
I would at least decouple the "valueMap" logic from here, if possible. There is already a TransformFunc concept somewhere we could reuse.

Reply:
Since what we are doing is flattening a label to the metric name, perhaps we could call it `FromValueWithLabelInName`?

Reply:
Sounds good, though that would be `FromLabelValueWithLabelInName` :smile:

Reply:
Changed, though I'm not sure if it's better, as Transform function require bunch of type assertions again.

Reply:
I think that together they are longer but also makes them slightly easier to read, in my view. I'd settle for the current approach.